src/app/features/toasts/use-toast-handlers.ts

```diff
@@ -0,0 +1,51 @@
+import { useCallback, useState } from 'react';
+
+import type { ToastProps } from '@app/ui/components/toast/toast';
+
+export function useToastHandlers() {
+  const [toasts, setToasts] = useState(new Map());
+
+  const handleAddToast = useCallback((toast: ToastProps) => {
+    setToasts(prev => {
+      const newMap = new Map(prev);
+      newMap.set(String(Date.now()), { ...toast });
+      return newMap;
+    });
+  }, []);
+
+  return {
+    toasts,
+    handleRemoveToast: useCallback((key: any) => {
+      setToasts(prev => {
+        const newMap = new Map(prev);
+        newMap.delete(key);
+        return newMap;
+      });
+    }, []),
+    handleDispatchInfo: useCallback(
+      (message: string) => handleAddToast({ message, variant: 'info' }),
+      [handleAddToast]
+    ),
+    handleDispatchSuccess: useCallback(
+      (message: string) => handleAddToast({ message, variant: 'success' }),
+      [handleAddToast]
+    ),
+    handleDispatchError: useCallback(
+      (message: string) => handleAddToast({ message, variant: 'error' }),
+      [handleAddToast]
+    ),
+    handleDispatchPromise: useCallback(
+      async (promise: Promise<any>, msgs: { success: string; error: string }) => {
+        return promise
+          .then(data => {
+            handleAddToast({ message: msgs.success, variant: 'success' });
+            return data;
+          })
+          .catch(() => {
+            handleAddToast({ message: msgs.error, variant: 'error' });
+          });
+      },
+      [handleAddToast]
+    ),
+  };
+}
```

Comment:
Isn't it better to return a single object with all these functions inside a `useMemo`, rather than 5x useCallbacks?